admin/lib/validator/check-window-v2/check-window-admin-end-date-validator.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+const moment = require('moment')
+
+const checkWindowErrorMessages = require('../../errors/check-window-v2')
+const dateService = require('../../../services/date.service')
+const XRegExp = require('xregexp')
+const { isEmpty, isInt } = require('validator')
+
+/**
+ * Validates check window admin end date data
+ * @param {Object} validationError
+ * @param {Object} adminEndDateData
+ */
+module.exports.validate = (validationError, adminEndDateData) => {
+  const currentDate = moment.utc()
+  const currentYear = currentDate.format('YYYY')
+  const adminEndDate = dateService.createUTCFromDayMonthYear(adminEndDateData.adminEndDay,
+    adminEndDateData.adminEndMonth, adminEndDateData.adminEndYear)
+  const maxDaysInMonth = adminEndDate ? moment(adminEndDate).daysInMonth() : 31
```

Comment:
Those should already be moment objects, doing `moment()` over them seems redundant (that happens in the other validators too) although this is not a ‘breaking’ thing?

Reply:
@danwilliams updated